web/src/message_viewport.js

```diff
@@ -33,6 +33,10 @@ function make_dimen_wrapper(dimen_name, dimen_func) {
 export const height = make_dimen_wrapper("height", $.fn.height);
 export const width = make_dimen_wrapper("width", $.fn.width);
 
+export function max_message_height() {
+    return document.querySelector("html").offsetHeight * 0.65;
+}
```

Comment:
Hmm, would it be better to make the `height` and `width` functions use the DOM API rather than whatever they're doing, instead of introducing a new function here? 

If not, we should add a comment explaining why the two different forms of calculation are appropriate.

(As a side note, the "This increases performance" comment in the commit message would be a lot nicer if it described what operation increases performance by 100x -- does it make the whole `condense_and_collapse` significantly faster? I would have expected this check, which is only run once, to be small compared to the rest)

Reply:
The only reason not to overhaul the `height` and `width` functions right away is to be a little bit conservative here.  I fully support changing those out as well, but it's not really in the scope of simply making condense/collapse easier to understand, and there's some risk.  At minimum if we want to proceed on that idea (and I think we should), we should do it as a separate commit.

Reply:
(The 'height` and `width` functions do get used by a bunch of other places in the codebase.)

Reply:
For this particular change (`max_message_height()`), the performance impact is actually extremely negligible in either direction, since condense/collapse only needs to get the viewport height once per resize event.  This change is all about code simplification.  (The other cache-related commit has a little bit more at stake, since we are dealing with multiple messages, but it's also a no-brainer to remove that cache, since the cache entries actually get invalidated after one read, which totally defeats the purpose of the cache.)

Reply:
If it's negligible, maybe we can wait and do all those height/width changes as a separate PR?

Reply:
No, I think the "negligible" part of my comment only applies to performance.  It's a significant code cleanup that we should while the iron is hot.  I think we just need to tweak the commit messages a bit.

I think for both messages, we can simply say that the main goal for the cleanups is to simplify the code, and then just say that the reason, in both cases, to remove the cache, is that we don't actually ever use the cache values more than once in any kind of correct way.  We can move this discussion back to chat if that helps to wordsmith the messages, or maybe Tim can just edit them.